src/core/search/autocompleteresponsetransformer.js

```diff
@@ -0,0 +1,37 @@
+import AutoCompleteData, { AutoCompleteResult } from '../models/autocompletedata';
+
+/**
+ * A data transformer that takes the response object from an
+ * AutoComplete request and transforms it into a front-end oriented
+ * data structure that our component library and core storage understand.
+ */
+export default class AutoCompleteResponseTransformer {
```

Comment:
Let's add some unit tests for this class.

Reply:
+2 for tests!